frontend/src/utils/AuthContext.tsx

```diff
@@ -207,13 +207,14 @@ export const AuthProvider = ({ children }: AuthProviderProps) => {
     return unsubscribe;
   }, [user, router, loading]);
 
-  if (loading || !isAuthenticated) {
-    return (
-      <div>
-        <Loading />
-      </div>
-    );
-  }
+  // No longer needed, moved to individual templates to avoid white flash on load
+  // if (loading || !isAuthenticated) {
+  //   return (
+  //     <div>
+  //       <Loading />
+  //     </div>
+  //   );
+  // }
```

Comment:
Also, login as a volunteer and try navigate to a protected route like event/create, you will notice the page flashes first (not ideal)

Reply:
So this is actually a surprisingly difficult issue. I spent a few hours researching possible alternatives, and there actually isn't a good solution. Even [professional websites](https://app.splitbee.io/projects/theodorusclarence.com) have the same exact issue we're facing.

The main problem with our current implementation is that having a white flash before every route navigation is very bad from a UI perspective. I tested on both my laptop and phone, and our current website has a half-second white flash over the entire page every time the user navigates to a new route, no exceptions. It makes our website feel very glitchy and unprofessional.

The reason this is happening is because route navigation is instant, but authentication always takes a split second before returning a result. The proper solution to this issue is to implement server-side authentication using the Next.js `middleware.ts` file (see [docs](https://nextjs.org/docs/app/building-your-application/routing/middleware)) so that we intercept route navigation and only return a route once authentication completes. However, I looked into it and implementing this change would require us to refactor too much, because currently Firebase authenticates on the client-side whereas `middleware.ts` requires server-side authentication. We would need to use `firebase-admin` and send our token to the server via cookies using the `nookies` package, etc.

The compromise I decided on was that we would leave authentication handled on the client side, but to make the loading flash appear as invisible as possible. So rather than always loading a white background before every route, the page skeleton loads first and the children load only if the user is actually authenticated. The flash is still there, but because our loading screen looks almost the same as our final result (minus the children), it's effectively invisible.

> Also, login as a volunteer and try navigate to a protected route like event/create, you will notice the page flashes first (not ideal)

Only the page layout flashes. None of the children are actually shown until authentication completes, so no sensitive information is actually shown.

It's not a perfect solution, but it minimizes the tradeoffs we need to make. With the current white loading screen, in-site navigation is very distracting / glitchy (due to the white flash). With this change, in-site navigation is seamless, but the downside is that sometimes the loading screen will seem "wrong' if you directly type in the protected route from the URL. However, users will be navigating with our in-site navigation (buttons + navbar) much more than they will be from the URL bar, so I think it's worth it.

The only proper solution without tradeoffs is to dismantle AuthContext and implement a proper middleware + cookies.

Reply:
Yep, that is all valid. I thought about this a lot too when we started implementing authentication. I pushed a commit to hide the navbar. Keeping the right of the Welcome template is a nice touch for navigation between authentication paths like /login /signup but since it shows the image first, in a case where someone navigates to one of the those paths when already logged in, the image shows up first before showing the right content. A little bit uneasy about that but this PR Should not be blocked any further because of that

Reply:
Edit: I removed the change to hide the navbar

Reply:
Yep, this is the same issue as someone who is not logged in will still see the background gradient if they navigate to a protected page. Unfortunately, there is no solution unless we want to implement a proper middleware.